### PR TITLE
Models for Privacy Center configuration

### PR DESCRIPTION
**Pull request by galvana**
Closes [PROD-1787](https://ethyca.atlassian.net/browse/PROD-1787)

### Description Of Changes

Adding `plus_privacy_center_config` table to store a minimal version of the Privacy Center `config.json`. Only a subset of the spec is saved to the database, enough for the Admin UI to generate access and erasure requests.

### Code Changes

* [ ] Added new `plus_privacy_center_config` table
* [ ] `PrivacyCenterConfig` model and schema
* [ ] Validation tests

### Steps to Confirm

* [ ] See https://github.com/ethyca/fidesplus/pull/1354

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

[PROD-1787]: https://ethyca.atlassian.net/browse/PROD-1787?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/fides/api/schemas/privacy_center_config.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    class Config:
+        extra = "ignore"
+
+
+class PrivacyCenterConfig(FidesSchema):
+    actions: List[PrivacyRequestOption]
+
+    class Config:
+        extra = "ignore"
```
Comment (pattisdr) [inline src/fides/api/schemas/privacy_center_config.py:43]: Are you sure?  It seems a bit surprising to me that you can send in a large payload and only a percentage of it is saved.

File: src/fides/api/models/privacy_center_config.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            "description_subtext": [
+                "In order to opt-out of certain consent preferences, we may need to identify your account via your email address. This is optional."
+            ],
+            "icon_path": "/consent.svg",
+            "identity_inputs": {"email": "optional"},
+            "title": "Manage your consent",
+            "modalTitle": "Manage your consent",
+            "confirmButtonText": "Continue",
+            "cancelButtonText": "Cancel",
+        },
```
Comment (pattisdr) [inline src/fides/api/models/privacy_center_config.py:53]: Why do we add all these sections in right now when we can't use them or save them?

File: src/fides/api/models/privacy_center_config.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+
+class PrivacyCenterConfig(Base):
+    @declared_attr
+    def __tablename__(self) -> str:
+        return "plus_privacy_center_config"
+
+    config = Column(
+        MutableDict.as_mutable(JSONB),
+        nullable=False,
```
Comment (pattisdr) [inline src/fides/api/models/privacy_center_config.py:105]: Nothing sensitive in here right? Doesn't seem to be -

File: src/fides/api/models/privacy_center_config.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                "When you use our services, you're trusting us with your information. We understand this is a big responsibility and work hard to protect your information and put you in control."
+            ],
+            "policy_key": "default_consent_policy",
+            "title": "Manage your consent",
+        },
+    },
+}
+
+
+class PrivacyCenterConfig(Base):
```
Comment (pattisdr) [inline src/fides/api/models/privacy_center_config.py:98]: I generally like to see at least one test on at the model level - if you end up moving your `create_or_update` logic here that would be a good thing to test!

**Review by pattisdr — COMMENTED** [review]
All comments are questions and/or minor opinions! good work, clean, easy to follow

File: src/fides/api/models/privacy_center_config.py (hunk 4 above)
Comment (galvana) [reply src/fides/api/models/privacy_center_config.py:27]: I added a test now that I have the `create_or_update` function on the model.

**Review by galvana — COMMENTED** [review]
Here are the fides changes, please see the related comments in fidesplus.

File: src/fides/api/schemas/privacy_center_config.py (hunk 1 above)
Comment (pattisdr) [reply src/fides/api/schemas/privacy_center_config.py:43]: ah okay that makes sense! thank you for explaining

**Review by pattisdr — APPROVED** [review]
🎉 nice one Adrian

**Pull request merged**